### PR TITLE
[BugFix] Fix bug Json metadata lost fn info

### LINKED ISSUE #28770: Stream load failed after FE restarted
### Steps to reproduce the behavior (Required)
1.  rm -rf fe/meta/*
2.  deploy StarRocks cluster of version 3.1.0-1778465
3.  CREATE TABLE `table_hitcount` (
  `databaseName` varchar(200) NULL COMMENT "数据库名",
  `tableName` varchar(200) NULL COMMENT "表名称",
  `queryTime` varchar(50) NULL COMMENT "查询时间",
  `queryId` varchar(50) NULL COMMENT "",
  `partitionHitSum` int(11) NULL COMMENT "查询分区数量",
  `partitionSum` int(11) NULL COMMENT "查询分区总数",
  `tabletHitNum` int(11) NULL COMMENT "查询tablet数量",
  `tabletSum` int(11) NULL COMMENT "查询分区中tablet总数",
  `startHitPartition` varchar(20) NULL COMMENT "查询开始分区",
  `dt` date NULL COMMENT "分区日期",
  `clusterAddress` varchar(50) NULL COMMENT "",
  `costTime` int(11) NULL COMMENT "",
  `tableQueryCount` int(11) NULL COMMENT ""
) ENGINE=OLAP 
DUPLICATE KEY(`databaseName`, `tableName`)
COMMENT "OLAP"
PARTITION BY date_trunc('day', dt)
DISTRIBUTED BY HASH(`databaseName`) BUCKETS 1 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"enable_persistent_index" = "false",
"replicated_storage" = "true",
"compression" = "LZ4"
4. stream load: success
5. Restart FE
6. stream load: failed

### Expected behavior (Required)
Stream load success after FE restarted

### Real behavior (Required)
Stream load failed
```
java.lang.IllegalStateException: null
        at com.google.common.base.Preconditions.checkState(Preconditions.java:496) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.analysis.FunctionCallExpr.isAggregateFunction(FunctionCallExpr.java:324) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr$1.apply(Expr.java:103) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr$1.apply(Expr.java:100) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr.isAggregate(Expr.java:833) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.FunctionCallExpr.toThrift(FunctionCallExpr.java:354) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr.treeToThriftHelper(Expr.java:807) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr.treeToThrift(Expr.java:762) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.Expr.treesToThrift(Expr.java:361) ~[starrocks-fe.jar:?]
        at com.starrocks.planner.OlapTableSink.createPartition(OlapTableSink.java:342) ~[starrocks-fe.jar:?]
        at com.starrocks.planner.OlapTableSink.complete(OlapTableSink.java:239) ~[starrocks-fe.jar:?]
        at com.starrocks.planner.OlapTableSink.complete(OlapTableSink.java:220) ~[starrocks-fe.jar:?]
        at com.starrocks.planner.StreamLoadPlanner.plan(StreamLoadPlanner.java:205) ~[starrocks-fe.jar:?]
        at com.starrocks.service.FrontendServiceImpl.streamLoadPutImpl(FrontendServiceImpl.java:1479) ~[starrocks-fe.jar:?]
        at com.starrocks.service.FrontendServiceImpl.streamLoadPut(FrontendServiceImpl.java:1433) ~[starrocks-fe.jar:?]
        at com.starrocks.thrift.FrontendService$Processor$streamLoadPut.getResult(FrontendService.java:3346) ~[starrocks-fe.jar:?]
        at com.starrocks.thrift.FrontendService$Processor$streamLoadPut.getResult(FrontendService.java:3326) ~[starrocks-fe.jar:?]
        at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38) ~[libthrift-0.13.0.jar:0.13.0]
        at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:38) ~[libthrift-0.13.0.jar:0.13.0]
        at com.starrocks.common.SRTThreadPoolServer$WorkerProcess.run(SRTThreadPoolServer.java:311) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_202]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_202]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_202]

```

### StarRocks version (Required)
 - 3.1.0-1778465

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #28770

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/catalog/ExpressionRangePartitionInfoTest.java [hunk 1]
```diff
@@ -419,4 +439,47 @@ public void testFixedRange5() throws DdlException, AnalysisException {
...
+                "tableName varchar(200) NULL COMMENT \"表名称\",\n" +
+                "queryTime varchar(50) NULL COMMENT \"查询时间\",\n" +
+                "queryId varchar(50) NULL COMMENT \"\",\n" +
+                "partitionHitSum int(11) NULL COMMENT \"查询分区数量\",\n" +
+                "partitionSum int(11) NULL COMMENT \"查询分区总数\",\n" +
+                "tabletHitNum int(11) NULL COMMENT \"查询tablet数量\",\n" +
+                "tabletSum int(11) NULL COMMENT \"查询分区中tablet总数\",\n" +
+                "startHitPartition varchar(20) NULL COMMENT \"查询开始分区\",\n" +
+                "dt date NULL COMMENT \"分区日期\",\n" +
+                "clusterAddress varchar(50) NULL COMMENT \"\",\n" +
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/catalog/ExpressionRangePartitionInfoTest.java:456]: please change this Chinese character to English

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ExpressionRangePartitionInfo.java|10|10|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**